app/helpers/filter_helper.rb

```diff
@@ -1,12 +1,16 @@
 module FilterHelper
   def filter_select(label: nil, scope:, collection:, key: :id, value: :name, selected: nil)
-    label ||= "Filter #{scope.to_s.gsub("_", ' ')}"
-    label_tag(label) + collection_select(:filters, scope, collection || {}, key, value, { include_blank: true, selected: selected }, class: "form-control")
+    sanitize_scope = scope.to_s.gsub("_", " ")
+    label ||= "Filter #{sanitize_scope}"
+    label_id = "filters_#{sanitize_scope}"
```

Comment:
This helper is reused across many pages. Are we sure we're never going to see the same scope reused e.g. across multiple forms in the same page? Would it make sense to just generate a UUID to use here instead?

Reply:
Potentially? I'm not sure it makes sense for that to happen unless you have multiple collections on the same page.

How would you suggest doing this. Collection select combines the object passed in with the method. In our case :filters + "by_source" or whatever, and that creates the name.  

From what I can tell if I use a uuid I have to pass it into the collection select 
![image](https://github.com/rubyforgood/human-essentials/assets/14540596/9d25847b-8572-423d-ab70-79864311bf7f)

It is going to get added to the name, which breaks the form submission. So that would require a controller change I think?

Reply:
I'm a bit confused here... the ID shouldn't affect the name. In the code above, the `label_id` is not being passed into the `collection_select` method?

Reply:
I did a poor job explaining. It doesn't matter I figured a way around the issue, I can just directly set the id value on both the label and the selection. Previously it was creating defaults using the object and the method passed in.

Reply:
```rb
id = "filters_#{scope}_#{SecureRandom.uuid}"
label_tag(id, label) + text_field(:filters, scope, class: "form-control", id: id, value: selected)
```

So if I do something like this. I get following input:
```html
<label for="filters_by_status_d7c5ede8-2953-4aeb-9d4e-c45d6efee8c5">Filter by status</label>
<select class="form-control" id="filters_by_status_d7c5ede8-2953-4aeb-9d4e-c45d6efee8c5" name="filters[by_status]">
```

So there won't be id issues anymore, however, the spec tests all based on the ids of the selectors. Ex: `select('Fulfilled', from: "filters_by_status")`. So all the systems specs would fail. I could change the specs to use name values instead of id ex: `select('Fulfilled', from: "filters[by_status]")`

Not sure what the best approach is.

Reply:
I think changing it to `filters[by_status]` should be fine.

I know it's kind of annoying, but if we're going to make this change, I'd rather do it in a way that doesn't introduce surprises when people use this filter without realizing there's a limitation (in that you couldn't use it more than once per page).

Having said that, in general I don't think accessibility has ever been a focus in the app, and if you look I'm sure you can find _lots_ of things to improve - it might even need to be a separate epic-level issue.

Reply:
Made the change. There were less instances than I expected, only 14 total.